taker-frontend/src/components/Trade.tsx

```diff
@@ -93,19 +99,22 @@ export default function Trade({
     let outerCircleBg = useColorModeValue("gray.100", "gray.700");
     let innerCircleBg = useColorModeValue("gray.200", "gray.600");
 
-    const referencePrice = `$${referencePriceAsNumber?.toLocaleString() || "0.0"}`;
-    const askPrice = `$${askPriceAsNumber?.toLocaleString() || "0.0"}`;
-    const liquidationPrice = `$${liquidationPriceAsNumber?.toLocaleString() || "0.0"}`;
+    // Use `Number` wrapper here to ensure the "," separator is printed.
+    const askPrice = `$${Number(askPriceAsNumber)?.toLocaleString() || "0.0"}`;
+    const liquidationPrice = `$${Number(liquidationPriceAsNumber)?.toLocaleString() || "0.0"}`;
 
     const { isOpen, onOpen, onClose } = useDisclosure();
 
     const margin = (quantity / parcelSize) * marginPerParcel;
 
-    const balanceTooLow = walletBalance < margin;
+    let btc_to_open_position = margin + opening_fee;
+
+    const balanceTooLow = walletBalance < btc_to_open_position;
     const quantityTooHigh = maxQuantity < quantity;
     const quantityTooLow = minQuantity > quantity;
     const quantityGreaterZero = quantity > 0;
     const quantityIsEvenlyDivisibleByIncrement = isEvenlyDivisible(quantity, parcelSize);
+    const hourlyFundingRate = (Number.parseFloat(funding_fee_annualized_percent) / 365 / 24).toFixed(5);
```

Comment:
Why aren't we doing this in the backend?

Reply:
oh, we actually are. That is legacy that should be changed!